impeller/aiks/experimental_canvas.cc

```diff
@@ -210,30 +212,148 @@ void ExperimentalCanvas::SaveLayer(
+
+    // If the pass context returns a backdrop texture, we need to draw it to the
+    // current pass. We do this because it's faster and takes significantly less
+    // memory than storing/loading large MSAA textures. Also, it's not possible
+    // to blit the non-MSAA resolve texture of the previous pass to MSAA
+    // textures (let alone a transient one).
+    Rect size_rect = Rect::MakeSize(restore_size);
+    auto msaa_backdrop_contents = TextureContents::MakeRect(size_rect);
+    msaa_backdrop_contents->SetStencilEnabled(false);
+    msaa_backdrop_contents->SetLabel("MSAA backdrop");
+    msaa_backdrop_contents->SetSourceRect(size_rect);
+    msaa_backdrop_contents->SetTexture(
+        rendering_config.inline_pass_context->GetTexture());
+
+    Entity msaa_backdrop_entity;
+    msaa_backdrop_entity.SetContents(std::move(msaa_backdrop_contents));
+    msaa_backdrop_entity.SetBlendMode(BlendMode::kSource);
+    msaa_backdrop_entity.SetClipDepth(std::numeric_limits<uint32_t>::max());
+    if (!msaa_backdrop_entity.Render(renderer_,
+                                     *render_passes_.back()
+                                          .inline_pass_context->GetRenderPass(0)
+                                          .pass)) {
+      VALIDATION_LOG << "Failed to render MSAA backdrop filter entity.";
+      return;
+    }
+
+    // Restore any clips that were recorded before the backdrop filter was
+    // applied.
+    auto& replay_entities = clip_coverage_stack_.GetReplayEntities();
+    for (const auto& replay : replay_entities) {
+      SetClipScissor(
+          clip_coverage_stack_.CurrentClipCoverage(),
+          *render_passes_.back().inline_pass_context->GetRenderPass(0).pass,
+          GetGlobalPassPosition());
+      if (!replay.entity.Render(renderer_,
+                                *render_passes_.back()
+                                     .inline_pass_context->GetRenderPass(0)
+                                     .pass)) {
+        VALIDATION_LOG << "Failed to render entity for clip restore.";
+      }
+    }
   }
 
   // When applying a save layer, absorb any pending distributed opacity.
   Paint paint_copy = paint;
   paint_copy.color.alpha *= transform_stack_.back().distributed_opacity;
   transform_stack_.back().distributed_opacity = 1.0;
 
+  // Backdrop Filter must expand bounds to at least the clip stack, otherwise
+  // the coverage of the parent render pass.
   Rect subpass_coverage = bounds->TransformBounds(GetCurrentTransform());
-  auto target =
-      CreateRenderTarget(renderer_,
-                         ISize::MakeWH(subpass_coverage.GetSize().width,
-                                       subpass_coverage.GetSize().height),
-                         1u, Color::BlackTransparent());
-  entity_pass_targets_.push_back(std::move(target));
+  if (backdrop_filter_contents) {
+    subpass_coverage = clip_coverage_stack_.CurrentClipCoverage().value_or(
+        save_layer_state_.back().coverage);
```

Comment:
IIUC from earlier, the plan is to do a `Save()` instead in cases where this ends up being zero size in order to avoid making a zero size texture/RenderPass?
Then, stuff appended within the "skipped" SaveLayer may still append draws, but they shouldn't be able to draw anything into the parent texture given the zero size scissor.

Reply:
This is done now. Also, on partial repaint examples we have fewer composited layers compared to the current entity pass.  I'm not sure why the current entity pass skip logic doesn't seem to work at eliding the entire layer when there are backdrops.

### Experimental Canvas

![image](https://github.com/flutter/engine/assets/8975114/2c4ee7f1-7356-4425-b606-e59ef908a9e1)


### Current 

![image](https://github.com/flutter/engine/assets/8975114/982b6945-1fef-48ab-892f-e1e71bc0598f)